### ISSUE TITLE
jenkins 2.2.122 - Unrecognized input header

### ISSUE THREAD

**Issue by cmoulliard** [fabric8io/fabric8#5818]
When we install the cd-pipeline pods, the jenkins pod isn't started and a CrashLoopBackOff is reported "oc get pods"

```
jenkins-ka8rt                        0/1       CrashLoopBackOff   1          1m

Log of Openshift

Mar 10 07:05:38 vagrant.f8 openshift[1019]: E0310 07:05:38.642331    1019 pod_workers.go:125] Error syncing pod 56c38326-e68e-11e5-ab95-08002778481b, skipping: failed to "StartContainer" for "jenkins" with RunContainerError: "failed to call event handler: Unrecognized input header"
Mar 10 07:05:38 vagrant.f8 openshift[1019]: I0310 07:05:38.677827    1019 manager.go:2079] Back-off 10s restarting failed container=jenkins pod=jenkins-ka8rt_default(56c38326-e68e-11e5-ab95-08002778481b)
Mar 10 07:05:38 vagrant.f8 openshift[1019]: E0310 07:05:38.695086    1019 pod_workers.go:125] Error syncing pod 56c38326-e68e-11e5-ab95-08002778481b, skipping: failed to "StartContainer" for "jenkins" with CrashLoopBackOff: "Back-off 10s restarting failed container=jenkins pod=jenkins-ka8rt_default(56c38326-e68e-11e5-ab95-08002778481b)"

Docker image used : fabric8/jenkins-docker:2.2.122
```

Pod info 

```
[vagrant@vagrant ~]$ oc describe pod jenkins-rdfta
Name:		jenkins-rdfta
Namespace:	default
Image(s):	fabric8/jenkins-docker:2.2.122
Node:		172.28.128.4/172.28.128.4
Start Time:	Thu, 10 Mar 2016 07:45:44 +0000
Labels:		group=io.fabric8.devops.apps,project=jenkins,provider=fabric8,version=2.2.122
Status:		Running
Reason:
Message:
IP:		172.17.0.2
Controllers:	ReplicationController/jenkins
Containers:
  jenkins:
    Container ID:	docker://7db4c1b3f593ad40dd8bd04ce202102b4245acd38036936bd518165ff0f92405
    Image:		fabric8/jenkins-docker:2.2.122
    Image ID:		docker://sha256:3c44b73179b052e7d7c3db7cd6c8489ea837b80e84ff38b518c5c839b73c4600
    QoS Tier:
      cpu:		BestEffort
      memory:		BestEffort
    State:		Terminated
      Reason:		Error
      Exit Code:	137
      Started:		Thu, 10 Mar 2016 07:45:48 +0000
      Finished:		Thu, 10 Mar 2016 07:46:18 +0000
    Ready:		False
    Restart Count:	0
    Environment Variables:
      DOMAIN:				vagrant.f8
      JENKINS_WORKFLOW_GIT_REPOSITORY:	https://github.com/fabric8io/jenkins-workflow-library.git
      PROJECT_VERSION:			2.2.122
      JENKINS_GOGS_PASSWORD:		RedHat$1
      JENKINS_GOGS_USER:		gogsadmin
      JENKINS_GOGS_EMAIL:		gogsadmin@fabric8.local
      FABRIC8_CONSOLE:			http://fabric8.vagrant.f8
      KUBERNETES_NAMESPACE:		default (v1:metadata.namespace)
Conditions:
  Type		Status
  Ready 	False
Volumes:
  data:
    Type:	HostPath (bare host directory volume)
    Path:	/var/run/docker.sock
  jenkins-docker-cfg:
    Type:	Secret (a secret that should populate this volume)
    SecretName:	jenkins-docker-cfg
  jenkins-token-nny2t:
    Type:	Secret (a secret that should populate this volume)
    SecretName:	jenkins-token-nny2t
Events:
  FirstSeen	LastSeen	Count	From			SubobjectPath			Type		Reason		Message
  ---------	--------	-----	----			-------------			--------	------		-------
  37s		37s		1	{default-scheduler }					Normal		Scheduled	Successfully assigned jenkins-rdfta to 172.28.128.4
  36s		36s		1	{kubelet 172.28.128.4}	spec.containers{jenkins}	Normal		Pulling		pulling image "fabric8/jenkins-docker:2.2.122"
  33s		33s		1	{kubelet 172.28.128.4}	spec.containers{jenkins}	Normal		Pulled		Successfully pulled image "fabric8/jenkins-docker:2.2.122"
  33s		33s		1	{kubelet 172.28.128.4}	spec.containers{jenkins}	Normal		Created		Created container with docker id 7db4c1b3f593
  33s		33s		1	{kubelet 172.28.128.4}	spec.containers{jenkins}	Normal		Started		Started container with docker id 7db4c1b3f593
  2s		2s		1	{kubelet 172.28.128.4}	spec.containers{jenkins}	Normal		Killing		Killing container with docker id 7db4c1b3f593: failed to call event handler: Unrecognized input header
  2s		2s		1	{kubelet 172.28.128.4}					Warning		FailedSync	Error syncing pod, skipping: failed to "StartContainer" for "jenkins" with RunContainerError: "failed to call event handler: Unrecognized input header"

  1s	1s	1	{kubelet 172.28.128.4}	spec.containers{jenkins}	Normal	Pulled	Container image "fabric8/jenkins-docker:2.2.122" already present on machine
  1s	1s	1	{kubelet 172.28.128.4}	spec.containers{jenkins}	Normal	Created	Created container with docker id 9faaf51caa5e
```

Remark : We can create a docker process

```
docker run -it -p 8080:8080 --name jenkins fabric8/jenkins-docker:2.2.122
Running from: /usr/share/jenkins/jenkins.war
webroot: EnvVars.masterEnvVars.get("JENKINS_HOME")
Jenkins home directory: /var/jenkins_home found at: EnvVars.masterEnvVars.get("JENKINS_HOME")

```

**Comment by cmoulliard** [comment]
I think that the problem comes from this message reported by docker `Content-Type specified (plain/text) must be 'application/json'"`

**Comment by cmoulliard** [comment]
openshift has been patched : https://github.com/openshift/origin/issues/7873

**Comment by jimmidyson** [comment]
I might downgrade docker in the vagrant box to 1.9.x which should solve that issue. Think docker 1.10 isn't property supported yet. Pretty bad of docker to introduce something that affects api consumers like this though...

**Comment by cmoulliard** [comment]
Are you gonna to release a new box this morning ?

**Comment by jimmidyson** [comment]
On top of everything else I need to do... yes...

**Comment by cmoulliard** [comment]
Info from James : https://github.com/docker/docker/issues/19950

**Comment by cmoulliard** [comment]
Problem solved by downgrading docker 1.10.x to Docker 1.9.1 within the Jimmy Centos Vagrant box

**Comment by jimmidyson** [comment]
Awesome!
